soh/soh/Enhancements/timesaver_hook_handlers.cpp

```diff
@@ -753,6 +759,41 @@ void TimeSaverOnActorInitHandler(void* actorRef) {
         });
     }
 
+    if (actor->id == ACTOR_BG_SPOT03_TAKI && Flags_GetEventChkInf(EVENTCHKINF_LEARNED_ZELDAS_LULLABY) &&
+        (INV_CONTENT(ITEM_OCARINA_TIME) == ITEM_OCARINA_TIME ||
+         INV_CONTENT(ITEM_OCARINA_FAIRY) == ITEM_OCARINA_FAIRY)) {
+        bgSpot03UpdateHook = GameInteractor::Instance->RegisterGameHook<GameInteractor::OnActorUpdate>([](void* innerActorRef) mutable {
+            Actor* innerActor = static_cast<Actor*>(innerActorRef);
+            if (innerActor->id == ACTOR_BG_SPOT03_TAKI && Flags_GetEventChkInf(EVENTCHKINF_LEARNED_ZELDAS_LULLABY) &&
+                (INV_CONTENT(ITEM_OCARINA_TIME) == ITEM_OCARINA_TIME ||
+                 INV_CONTENT(ITEM_OCARINA_FAIRY) == ITEM_OCARINA_FAIRY) &&
+                (IS_RANDO || CVarGetInteger(CVAR_ENHANCEMENT("TimeSavers.SleepingWaterfall"), IS_RANDO))
```

Comment:
Not quite sure how I feel about it being default on for rando, either, but that's not blocking for me. I'd have to hear how others feel.

Reply:
I would prefer it to not be forced on for rando. I'm sure quite a lot of people will keep it on, but I imagine some people, especially in a race scenario, would appreciate the nuance that having this off would bring. For instance, it may be faster to clip past the waterfall with a Cucco or Hoverboots than playing the song, even if you have an Ocarina and the song handy. Maybe even the Mega-sidehop would be faster if you can set it up quickly enough. I think there's enough nuance between the convenience of playing the song vs tricks you can do potentially being faster that it's worth having the ability to turn it off in a rando run. Maybe there's merit to having this on but only for Ocarina of Time but that would take a lot more discussion. For now I'd merge this with it not being forced on for rando.

Reply:
I would surmise that the best option here is to have the checkbox in Enhancements that only affects vanilla, and a separate option that only affects rando (which could be configured to open the waterfall at any time OR only if ZL can be played OR never unless it is played).

The separate `IS_RANDO` check is something I've seen in several other places in this file. It could very well be redundant, but if it is, it should be removed in all instances.

Reply:
No, it was only in your specific case because IS_RANDO was accounted for in the CVarGet itself. Since I think we've decided that it should default on in rando, then just removing the check and only relying on the CVar is appropriate.

Reply:
Though, tbf, if there were other instances of IS_RANDO being used as the default for a CVarGet that were paired with a separate IS_RANDO check, those should be changed, too. I was just saying that the redundancy was specific.